mmrotate/datasets/dior.py

```diff
@@ -0,0 +1,239 @@
+                 version='oc',
+                 xmltype='obb',
+                 **kwargs):
+        assert xmltype in ['hbb', 'obb']
+        self.xmltype = xmltype
+        self.img_subdir = img_subdir
+        self.ann_subdir = ann_subdir
+        self.version = version
+        self.cat2label = {cat: i for i, cat in enumerate(self.CLASSES)}
+        super(DIORDataset, self).__init__(ann_file, pipeline, **kwargs)
+
+    def load_annotations(self, ann_file):
+        """Load annotation from XML style ann_file.
+
+        Args:
+            ann_file (str): Path of Imageset file.
+
+        Returns:
+            list[dict]: Annotation info from XML file.
+        """
+
+        data_infos = []
+        img_ids = mmcv.list_from_file(ann_file)
+        for img_id in img_ids:
+            data_info = {}
+
+            filename = osp.join(self.img_subdir, f'{img_id}.jpg')
+            data_info['filename'] = filename
+            xml_path = osp.join(self.img_prefix, self.ann_subdir,
+                                f'{img_id}.xml')
+            tree = ET.parse(xml_path)
+            root = tree.getroot()
+
+            size = root.find('size')
+            if size is not None:
+                width = int(size.find('width').text)
+                height = int(size.find('height').text)
+
+            if width is None or height is None:
+                img_path = osp.join(self.img_prefix, filename)
+                img = Image.open(img_path)
+                width, height = img.size
+            data_info['width'] = width
+            data_info['height'] = height
+            data_info['ann'] = {}
+            gt_bboxes = []
+            gt_labels = []
+            gt_polygons = []
+            gt_bboxes_ignore = []
+            gt_labels_ignore = []
+            gt_polygons_ignore = []
+
+            for obj in root.findall('object'):
+                cls = obj.find('name').text.lower()
+                label = self.cat2label[cls]
+                if label is None:
+                    continue
+
+                if self.xmltype == 'obb':
+                    bnd_box = obj.find('robndbox')
+                    polygon = np.array([
+                        float(bnd_box.find('x_left_top').text),
+                        float(bnd_box.find('y_left_top').text),
+                        float(bnd_box.find('x_right_top').text),
+                        float(bnd_box.find('y_right_top').text),
+                        float(bnd_box.find('x_right_bottom').text),
+                        float(bnd_box.find('y_right_bottom').text),
+                        float(bnd_box.find('x_left_bottom').text),
+                        float(bnd_box.find('y_left_bottom').text),
+                    ]).astype(np.float32)
+                else:
+                    bnd_box = obj.find('bndbox')
+                    if bnd_box is None:
+                        continue
+                    polygon = np.array([
+                        float(bnd_box.find('xmin').text),
```

Comment:
Dior may have some invalid objects. shall we filter them out here?

Reply:
How to filter invalid objects?

Reply:
I found we have filtered box whose width or height is smaller than 2 in ``poly2obb_np``.

https://github.com/open-mmlab/mmrotate/blob/6eb7a277ed62f64952587c9e41498f8e3c0cfe63/mmrotate/core/bbox/transforms.py#L347-L348